frontend/document/document.lua

```diff
@@ -39,6 +39,8 @@ function Document:new(from_o)
     self.__index = self
     if o._init then o:_init() end
     if o.init then o:init() end
+    -- Make sure the checksum has been generated before any document saving operations.
+    o:fastDigest()
```

Comment:
It's partial digest for no more than 10KB for any document file thus won't be of much overhead. And it will only be calculated for once for each document.